src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1860,21 +1860,37 @@ else if ("test".equals(configName)) {
     }
 
     private static void validateChapterWiseTesting(Iterator<Node> itrChecks,
-          Iterator<Node> itrConfigs, Set<String> styleChecks, String styleName, String ruleName) {
+          Iterator<Node> itrSample, Set<String> styleChecks, String styleName, String ruleName) {
         boolean hasChecks = false;
+        final Set<String> checkLinks = new HashSet<>();
 
         while (itrChecks.hasNext()) {
             final Node module = itrChecks.next();
             final String moduleName = module.getTextContent().trim();
             final String href = module.getAttributes().getNamedItem("href").getTextContent();
             final boolean moduleIsCheck = href.startsWith("checks/");
 
+            final String partialConfigUrl = "https://github.com/search?q="
+                    + "path%3Asrc%2Fmain%2Fresources%20path%3A**%2F" + styleName;
+
             if (!moduleIsCheck) {
+                if (href.startsWith(partialConfigUrl)) {
+                    assertWithMessage("google_style.xml rule '" + ruleName + "' module '"
+                            + moduleName + "' has too many config links").fail();
+                }
                 continue;
             }
```

Comment:
this is to check if there's extra config url, like:

* {Check name} (config) (config)

or there's only config url but no Check link.

Reply:
for this I had to divide the `expectedUrl` variable into 2 parts: `partialConfigUrl` & `wholeConfigUrl` so we can easily detect the config url